x-pack/plugins/apm/common/runtime_types/merge/index.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import * as t from 'io-ts';
+import { merge as lodashMerge } from 'lodash';
+import { isLeft } from 'fp-ts/lib/Either';
+import { ValuesType } from 'utility-types';
+
+export type MergeType<
+  T extends t.Any[],
+  U extends ValuesType<T> = ValuesType<T>
+> = t.Type<U['_A'], U['_O'], U['_I']> & {
+  _tag: 'MergeType';
+  types: T;
+};
+
+// this is similar to t.intersection, but does a deep merge
+// instead of a shallow merge
+
+export function merge<A extends t.Mixed, B extends t.Mixed>(
```

Comment:
Should it be called `deepMerge` then? 
Btw. I'm surprised this doesn't already exist in the shared typescript utils.

Reply:
it's `merge` because it uses `lodash.merge`. I figured `deepMerge` would be overkill.